components/style/properties/longhand/box.mako.rs

```diff
@@ -1175,9 +1173,9 @@ ${helpers.predefined_type("scroll-snap-coordinate",
                 m32: self.m32.to_computed_value(context),
                 m33: self.m33.to_computed_value(context),
                 m34: self.m34.to_computed_value(context),
-                m41: self.m41.to_computed_value(context),
-                m42: self.m42.to_computed_value(context),
-                m43: self.m43.to_computed_value(context),
+                m41: length_to_computed_number(self.m41.to_computed_value(context)),
```

Comment:
We already allow percentages in `translate`; these are resolved against the refbox in layout time. Interpolation fails for us here, and gecko uses interpolatematrix to handle these -- it defers interpolation till layout time. This is an independent issue we need to fix.

Yes, lengths can map to numbers by pixel value here, but percentages do not map.

(I think I've mentioned it before -- I'm quite annoyed that the spec doesn't really specify units here, but to be 100% correct, the homogeneous coordinates in the matrix should be in units of pixel, and the rest are unitless ratios, but it implicitly expects us to convert the ratio to pixels of the same numeric value)